packages/components/src/components/CodeEditor/languageSupport/squiggle.grammar

```diff
@@ -83,7 +83,15 @@ expressionWithoutParens[@isGroup="Expression"] {
             >
         "}"
     }
-    | Lambda { "{" ArgsOpen { "|" } LambdaArgs "|" blockContent "}" }
+    | Lambda {
+        "{"
+            (
+                ArgsOpen { "|" } LambdaArgs "|"
+                | "||" // special case - '||' would be parsed as a LogicOp token without this rule
```

Comment:
This fixes the first part of https://github.com/quantified-uncertainty/squiggle/issues/3336#issuecomment-2321478351.